kaldb/src/main/java/com/slack/kaldb/writer/SpanFormatter.java

```diff
@@ -103,80 +96,65 @@ public static Trace.KeyValue convertKVtoProto(
       try {
         switch (schema.getFieldsMap().get(key).getType()) {
           case KEYWORD -> {
-            tagBuilder.setVType(Trace.ValueType.STRING);
             tagBuilder.setFieldType(Schema.SchemaFieldType.KEYWORD);
             tagBuilder.setVStr(value.toString());
           }
           case TEXT -> {
-            tagBuilder.setVType(Trace.ValueType.STRING);
             tagBuilder.setFieldType(Schema.SchemaFieldType.TEXT);
             tagBuilder.setVStr(value.toString());
           }
           case IP -> {
-            tagBuilder.setVType(Trace.ValueType.STRING);
             tagBuilder.setFieldType(Schema.SchemaFieldType.IP);
             tagBuilder.setVStr(value.toString());
           }
           case DATE -> {
-            tagBuilder.setVType(Trace.ValueType.STRING);
             tagBuilder.setFieldType(Schema.SchemaFieldType.DATE);
-            tagBuilder.setVStr(value.toString());
+            tagBuilder.setVInt64(Instant.parse(value.toString()).toEpochMilli());
```

Comment:
Does date always come through with this format? I believe you can produce "date" types with different formats correct? https://www.elastic.co/guide/en/elasticsearch/reference/current/date.html

Reply:
that is indeed correct. Let me address that in a seprate PR i.e expose a `format` field.